internal/xsql/sqlValidator.go

```diff
@@ -115,3 +115,65 @@ func isSRFExists(node ast.Node) bool {
 	})
 	return exists
 }
+
+func validateFields(stmt *ast.SelectStatement) {
+	streamName := getStreamNames(stmt)
+	for i, field := range stmt.Fields {
```

Comment:
Not only fields must be revisited, the on condition in joins should also be revisited.